### PR TITLE
fix: search style

### PR DESCRIPTION
**Pull request by Hongbusi**
Appearance after one click：

image

image

### CODE REVIEW

**Comment by kiaking** [comment]
I can't reproduce this. What browser are you using...?

**Comment by Hongbusi** [comment]
> I can't reproduce this. What browser are you using...?

Chrome v108 (arm64)

**Comment by kiaking** [comment]
I'm using the same version but still can't see this happening 🤔 

Before Click
Screen Shot 2022-12-16 at 11 06 46

After Click
Screen Shot 2022-12-16 at 11 07 17

Can anyone else reproduce this?

**Comment by Hongbusi** [comment]
Let friends try it, some can be reproduced, and some can't.

But my access to vitepress is normal.

**Comment by Jinjiang** [comment]
@kiaking just for more information. The discussion in #78 is quite similar to this one. However, I can't reproduce either of them. For that PR, I just revamped and restructured the CSS code, not targeting any bugs. Maybe there are some edge cases we haven't touched, but I recommend moving forward after having comprehensive reproductions and details this time. Thanks.

**Comment by kiaking** [comment]
Yeah. Unless we can reproduce this, we can't really fix it. So let's wait to see if anyone can give us feedback on this. And if we don't get it, I think we should just close it.

**Comment by Jinjiang** [comment]
Agree. I think we can leave this PR open for several days until we receive more information to reproduce it. And once closed, it's still welcome to reopen/recreate.
Thanks.

**Comment by Jinjiang** [comment]
@kiaking I made it reproduced with @Justineo's help. Anyway, in short, the pre-condition is setting the "minimum font size" in Chrome with a non-zero value, which I guess is `12px` by default if you choose Chinese as the default locale. That explains how people met the bug but we couldn't reproduce it (my default locale setting is English).

Then I'd like to reopen this PR and retake a look at it. Hope you folks don't mind.

Thanks.

File: src/vitepress/components/VPNavBarSearch.vue [hunk 1]
```diff
@@ -50,6 +50,7 @@ function load() {
       <button
         type="button"
         class="DocSearch DocSearch-Button"
+        preload
```
Comment (Justineo) [inline src/vitepress/components/VPNavBarSearch.vue:53]: Maybe `data-preload`?

File: src/vitepress/components/VPNavBarSearch.vue [hunk 2]
```diff
@@ -227,26 +228,14 @@ function load() {
...
   transition: color 0.5s;
 }
-.DocSearch-Button:hover .DocSearch-Button-Key:first-child::after,
-.DocSearch-Button:hover .DocSearch-Button-Key:last-child {
+.DocSearch-Button:hover .DocSearch-Button-Key {
   color: var(--vt-c-brand-light);
 }
+
+.DocSearch-Button[preload] .DocSearch-Button-Key:first-child::after {
+  content: v-bind(metaKey);
```
Comment (Justineo) [inline src/vitepress/components/VPNavBarSearch.vue:239]: I also found that the default `metaKey` value (`'Meta'`) will cause a layout shift. Any idea to fix this?

Comment (Jinjiang) [reply src/vitepress/components/VPNavBarSearch.vue]: I tried a much-simplified way. What do you think about it now?

File: src/vitepress/components/VPNavBarSearch.vue [hunk 3]
```diff
@@ -13,15 +13,15 @@ const VPAlgoliaSearchBox = defineAsyncComponent(
 // payload), we delay initializing it until the user has actually clicked or
 // hit the hotkey to invoke it
 const loaded = ref(false)
-const metaKey = ref(`'Meta'`)
+const metaKey = ref(`Meta`)
```
Comment (Hongbusi) [inline src/vitepress/components/VPNavBarSearch.vue:16]:
```suggestion
const metaKey = ref('Meta')
```

File: src/vitepress/components/VPNavBarSearch.vue [hunk 4]
```diff
@@ -13,15 +13,15 @@ const VPAlgoliaSearchBox = defineAsyncComponent(
...
 
 onMounted(() => {
   if (!config.value.algolia) return
 
   // meta key detect (same logic as in @docsearch/js)
   metaKey.value = /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform)
-    ? `'⌘'`
-    : `'Ctrl'`
+    ? `⌘`
+    : `Ctrl`
```
Comment (Hongbusi) [inline src/vitepress/components/VPNavBarSearch.vue:24]:
```suggestion
    ? '⌘'
    : 'Ctrl'
```

**Review by Hongbusi — COMMENTED** [review]
Is this better?

File: src/vitepress/components/VPNavBarSearch.vue (hunk 2 above)
Comment (Jinjiang) [reply src/vitepress/components/VPNavBarSearch.vue]: Update: I moved the button into a `<ClientOnly>` component. So it won't be pre-rendered in the server side. That also means the hotkey will only show up after we already know which OS it's on.

**Comment by Jinjiang** [comment]
Since no more points for a while, I'd like to merge it if you folks don't mind. Further discussion is always welcome by a new issue or PR. Thanks.

**Pull request merged**